packages/landing/src/attribution.ts

```diff
@@ -0,0 +1,205 @@
+import { Extent } from 'ol/extent';
+import OlGeometryLayout from 'ol/geom/GeometryLayout';
+import OlPolygon from 'ol/geom/Polygon';
+import OlMap from 'ol/Map';
+import MapEventType from 'ol/MapEventType';
+import { transformExtent } from 'ol/proj';
+import Source from 'ol/source/Source';
+import { MapOptions, MapOptionType, WindowUrl } from './url';
+
+class AttributionBounds {
+    collection: AttributionCollection;
+    boundaries: OlPolygon[] = [];
+    minZoom: number;
+    maxZoom: number;
+    bbox: BBox;
+
+    constructor(collection: AttributionCollection) {
+        this.collection = collection;
+        const zoom = collection.summaries['linz:zoom'];
+        this.minZoom = zoom.min;
+        this.maxZoom = zoom.max;
+        this.bbox = collection.extent.spatial.bbox[0];
+    }
+
+    intersects(extent: Extent, zoom: number): boolean {
+        if (zoom > this.maxZoom || zoom < this.minZoom || !Wgs84.intersects(extent, this.bbox)) {
+            return false;
+        }
+
+        if (extent[0] < extent[2]) {
+            return this.boundaries.find((p) => p.intersectsExtent(extent)) != null;
+        }
+
+        const ext1: Extent = [extent[0], extent[1], extent[2] + 360, extent[3]];
+        if (this.boundaries.find((p) => p.intersectsExtent(ext1)) != null) {
+            return true;
+        }
+        ext1[2] = extent[2];
+        ext1[0] -= 360;
+
+        return this.boundaries.find((p) => p.intersectsExtent(ext1)) != null;
+    }
+}
+
+function convertToOlAttribution(stac: AttributionStac): AttributionBounds[] {
+    const colMap = new Map<string, AttributionBounds>();
+    const result: AttributionBounds[] = [];
+
+    for (const collection of stac.collections) {
+        const olattr = new AttributionBounds(collection);
+        result.push(olattr);
+        colMap.set(collection.id, olattr);
+    }
+
+    for (const f of stac.features) {
+        const col = colMap.get(f.collection ?? '');
+        if (col == null) {
+            throw new Error('Could not match feature to collection: ' + f.collection);
+        }
+        if (f.geometry.type === 'Polygon') {
+            col.boundaries.push(new OlPolygon(f.geometry.coordinates, OlGeometryLayout.XY));
+        } else if (f.geometry.type === 'MultiPolygon') {
+            for (const poly of f.geometry.coordinates) {
+                col.boundaries.push(new OlPolygon(poly, OlGeometryLayout.XY));
+            }
+        }
+    }
+
+    return result;
+}
+
+async function fetchAttributions(url: string): Promise<AttributionBounds[] | null> {
+    try {
+        const resp = await fetch(url);
+        if (resp.status < 300) {
+            return convertToOlAttribution((await resp.json()) as AttributionStac);
+        } else {
+            throw new Error(`fetch attribution failed [${resp.status}] ${resp.statusText}`);
+        }
+    } catch (err) {
+        console.error(err);
+        return null;
+    }
+}
+
+function sameContents(a: any[], b: any[]): boolean {
```

Comment:
this wouldnt work for objects, did you want to remove the any and  make it `null | string | number` ?

Reply:
It does work for objects because the are guaranteed to be the same objects. However since this is non obvious I will not use it and instead compare the rendered string which it probably quicker to compute anyway.